### PR TITLE
Add span smoke tests

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Also refactor the status of the instrumentation to be returned by instrument method. It avoids to send a probe status INSTALLED first then received later the ERROR one.
Report a probe status error when trying to instrument a single line span probe.

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/ConfigurationUpdater.java [hunk 1]
```diff
@@ -182,6 +182,9 @@ private void installNewDefinitions() {
 
   private void recordInstrumentationProgress(
       ProbeDefinition definition, InstrumentationResult instrumentationResult) {
+    if (instrumentationResult.isError()) {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/ConfigurationUpdater.java:185]: I find this contact kind of weird, with isError() not report a messages. 
1. does this is a mixed responsibility for classes? 
2. how does the error status message get to the sink?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/SpanInstrumentor.java [hunk 2]
```diff
@@ -94,25 +93,29 @@ private InsnList createSpan(LabelNode initSpanLabel) {
     return insnList;
   }
 
-  private void addRangeSpan(LineMap lineMap) {
+  private InstrumentationResult.Status addRangeSpan(LineMap lineMap) {
     Where.SourceLine[] targetLines = definition.getWhere().getSourceLines();
     if (targetLines == null || targetLines.length == 0) {
-      // no line capture to perform
-      return;
+      reportError("Missing line(s) in probe definition.");
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/SpanInstrumentor.java:99]: Maybe consider change the return value of instrument function?
I find it wried that we have both reportError and return InstrumentationResult.Status.

**Review by shatzi — APPROVED** [review]
Great we have integration tests for spans.

**Comment by shatzi** [comment]
After talking about it, refactor the internal logic would happen on later PR, this PR goal is to create a set of integration tests to validate the logic of the current code.

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/ConfigurationUpdater.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/ConfigurationUpdater.java:185]: Yep will refactor it in a next PR to centralize the handling

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/SpanInstrumentor.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/SpanInstrumentor.java:99]: sure, will do in the next refactoring PR

**Comment by jpbempel** [comment]
Merge as is, but will refactor the diagnostic messages handling in a next PR

**Pull request merged**